src/SConscript

```diff
@@ -30,36 +32,54 @@ def generate_daos_version_header(env):
     out = env.Substfile(tmpl_hdr_in, SUBST_DICT=sd)
     print("generated daos version header file: {}".format(out[0].abspath))
 
+def check_install_header(env, header):
+    """Runs a builder to ensure the header can standalone compile and installs it"""
+    result = env.CheckHeader(header)
```

Comment:
Previously CheckHeader was being called with a config, but now you're calling it with env?

Reply:
And in fact looking at the build logs I'm not sure that it's still running the check?

Reply:
Yes.   The issue here is SCons bug https://github.com/SCons/scons/issues/2757

When Configure is used on our internal headers, it resets the changed state so the rest of the code doesn't know our internal headers changed.   So, I worked around it.